pkg/assets/builder.go

```diff
@@ -75,13 +79,28 @@ func (a *AssetBuilder) remapImage(image string) (string, error) {
 		// 3. make kops and create/apply cluster
 		override := os.Getenv("DNSCONTROLLER_IMAGE")
 		if override != "" {
-			image = override
+			asset.Mirror = override
+			a.Assets = append(a.Assets, asset)
+			return image, nil
 		}
 	}
 
-	asset.Mirror = image
+	if strings.HasPrefix(image, GCR_IO) {
+		override, err := GetGoogleImageRegistryContainer(a.ClusterSpec, image)
```

Comment:
Again not really sure why these two cases are different (aka why is gcr.io special), but I'm sure I'll find out soon enough!